src/processdata.c

```diff
@@ -1475,19 +1475,56 @@ int DataDumpH5(Link** sys, unsigned int N, int* assignments, UnivVars* GlobalVar
     unsigned int i;
     unsigned int res = 0;
 
+
+
     if (my_rank == 0)	//Creating the file
     {
         char dump_loc_filename[ASYNCH_MAX_PATH_LENGTH];
         if (suffix)
-        {
-            char basename[ASYNCH_MAX_PATH_LENGTH];
-            strcpy(basename, GlobalVars->dump_loc_filename);
-
-            char *dot = strrchr(basename, '.');
-            if (dot != NULL)
-                *dot = '\0';
-
-            snprintf(dump_loc_filename, ASYNCH_MAX_PATH_LENGTH, "%s_%s.h5", basename, suffix);
+        {	//get the base name from the global file. parse the runID (the part before underscore)
+			unsigned int _index=0;
+			 _index= (int)(strchr(GlobalVars->dump_loc_filename, '_') - GlobalVars->dump_loc_filename); //position of the underscore char _
+
+			 if (_index > 0) {
+				 char baseName[ASYNCH_MAX_PATH_LENGTH];
+				 char baseTimestamp[11];
+				 strncpy(baseName, GlobalVars->dump_loc_filename, _index);
+				 baseName[_index] = '\0';
+				 strncpy(baseTimestamp, GlobalVars->dump_loc_filename + _index + 1, 10);
+				 baseTimestamp[10] = '\0';
+				 char *endptr;
+				 int timestamp = strtol(baseTimestamp, &endptr, 10);
+				 if (*endptr != '\0')
+				 {
+					 printf("Error: Output file name %s not standard: runID_timestamp.h5", GlobalVars->dump_loc_filename);
+					 res = 1;
+					 MPI_Bcast(&res, 1, MPI_UNSIGNED, 0, MPI_COMM_WORLD);
```

Comment:
You cannot `MPI_Bcast` here, this code is executed on rank == 0 only. This would stall the app.